tests/unit/mainwindow/test_tabwidget.py

```diff
@@ -39,6 +39,14 @@ def widget(self, qtbot, monkeypatch, config_stub):
                             usertypes.Backend.QtWebKit)
         return w
 
+    @pytest.fixture
+    def browser(self, qtbot, monkeypatch, config_stub):
```

Comment:
Does this need to be a real `TabbedBrowser`? There's a `tabbed_browser_stubs` fixture which creates two fake `TabbedBrowser` objects (`TabbedBrowserStub`) used elsewhere.

Reply:
The reason I created a real TabbedBrowser here is mainly because I wanted to performance test the TabbedBrowser addTab and shutdown methods. If I converted it to a stub, I would be testing the stub's addTab/shutdown methods, which isn't very useful :P. 

I actually could get away with a tabwidget here, but tabwidget dosen't have a 'shutdown' method, so I would need to re-implement it in the test, but it's pretty simple so I don't think that's much of an issue.

Does switching to a tabwidget seem like a better approach here?

Reply:
Oh, right! 😆 Seems fine to keep it as-is. We might need to revisit it after #600 (`TabbedBrowser` really shouldn't inherit from `TabWidget`), but for now, let's keep it.